server/mongo_data_access.go

```diff
@@ -226,7 +274,7 @@ func (dal *mongoDataAccessLayer) ConditionalDelete(query search.Query) (count in
 
 					if elementInSlice(id, successfulIds) {
 						// This resource was confirmed deleted
-						dal.invokeInterceptors("Delete", resourceType, elem.Resource)
+						dal.invokeInterceptorsAfter("Delete", resourceType, elem.Resource)
```

Comment:
Shouldn't you also have an `else` that invokes `OnError` (since it `else` would mean it was not successful)?  In that case, I think you'd need to create your own `error` (since you don't have one passed to you).